docs/resources/server/advancements.md

```diff
@@ -0,0 +1,173 @@
+
+The registry name of the trigger is supplied to `#getId`. This should match the registry name supplied to the trigger instance.
+
+A trigger instance is created via `#createInstance`. This method reads a criteria from JSON.
+
+```java
+@Override
+public ExampleTriggerInstance createInstance(JsonObject json, EntityPredicate.Composite player, DeserializationContext context) {
+  // Read conditions from JSON: item
+  return new ExampleTriggerInstance(player, item);
+}
+```
+
+Finally, a method is defined to check all trigger instances and run the listeners if their condition is met. This method takes in the `ServerPlayer` and whatever other data defined by the matching method in the `AbstractCriterionTriggerInstance` subclass. This method should internally call `SimpleCriterionTrigger#trigger` to properly handle checking all listeners. Most trigger instances call this method `#trigger`.
+
+```java
+// This method is unique for each trigger and is as such not overridden
+public void trigger(ServerPlayer player, ItemStack stack) {
+  this.trigger(player,
+    // The condition checker method within the AbstractCriterionTriggerInstance subclass
+    triggerInstance -> triggerInstance.matches(stack)
+  );
+}
+```
+
+Afterwards, an instance should be registered using `CriteriaTriggers#register` during `FMLCommonSetupEvent`.
+
+!!! important
+    `CriteriaTriggers#register` must be enqueued to the synchronous work queue via `FMLCommonSetupEvent#enqueueWork` as the method is not thread-safe.
+
+### Calling the Trigger
+
+Whenever the action being checked is performed, the `#trigger` method defined by the `SimpleCriterionTrigger` subclass should be called.
+
+```java
+// In some piece of code where the action is being performed
+// Where EXAMPLE_CRITERIA_TRIGGER is the custom criteria trigger
+public void performExampleAction(ServerPlayer player, ItemStack stack) {
+  // Run code to perform action
+  EXAMPLE_CRITERIA_TRIGGER.trigger(player, stack);
+}
+```
+
+Advancement Rewards
+-------------------
+
+When an advancement is completed, rewards may be given out. These can be a combination of experience points, loot tables, recipes for the recipe book, or a [function] executed as a creative player.
+
+```js
+// In some advancement JSON
+"rewards": {
+  "experience": 10,
+  "loot": [
+    "minecraft:example_loot_table",
+    "minecraft:example_loot_table2"
+    // ...
+  ],
+  "recipes": [
+    "minecraft:example_recipe",
+    "minecraft:example_recipe2"
+    // ...
+  ],
+  "function": "minecraft:example_function"
+}
+```
+
+Conditional Advancements
+------------------------
+
+Advancements can be [loaded conditionally and defaulted][conditional] depending on what information is present (mod loaded, item exists, etc.).
```

Comment:
Perhaps this should be included in the lead section instead of added at the very end of the article?

Reply:
I don't see why not. I typically include them at the end since I order the forge added features in the way I come up with them.